.github/workflows/auto-update.js

```diff
@@ -1,135 +1,172 @@
-      http_request(
-          "https://monitoring.nixos.org/prometheus/api/v1/query?query=channel_revision")
-          .then((res) => res.data.result)
-          .then(
-              (rev) => [rev.find((d) => d.metric.channel === "nixos-unstable"),
-                        rev.find((d) => d.metric.channel ===
-                                        "nixos-unstable-small"),
-  ]);
+    http_request(
+      "https://monitoring.nixos.org/prometheus/api/v1/query?query=channel_revision"
+    )
+      .then((res) => res.data.result)
+      .then((rev) => [
+        rev.find((d) => d.metric.channel === "nixos-unstable"),
+        rev.find((d) => d.metric.channel === "nixos-unstable-small"),
+      ]);
 
   function get_sha256(url) {
     return new Promise((resolve, reject) => {
-      exec(`nix-prefetch-url --type sha256 --unpack ${url}`,
-           (error, stdout, stderr) => {
-             if (error) {
-               return reject(error);
-             }
-
-             const lines = stdout.trim().split("\n");
-             resolve(lines[0]);
-           });
+      exec(
+        `nix-prefetch-url --type sha256 --unpack ${url}`,
+        (error, stdout, stderr) => {
+          if (error) {
+            return reject(error);
+          }
+
+          const lines = stdout.trim().split("\n");
+          resolve(lines[0]);
+        }
+      );
     });
   }
 
   const get_newest = (revisions) => {
-    const urls =
-        revisions.map((revision) => revision.metric.revision)
-            .map((commit_sha) =>
-                     `https://api.github.com/repos/nixos/nixpkgs/commits/${
-                         commit_sha}`);
-
-    return Promise.all(urls.map(http_request)).then(([ big, small ]) => {
+    const urls = revisions
+      .map((revision) => revision.metric.revision)
+      .map(
+        (commit_sha) =>
+          `https://api.github.com/repos/nixos/nixpkgs/commits/${commit_sha}`
+      );
+
+    return Promise.all(urls.map(http_request)).then(([big, small]) => {
       const big_date = new Date(big.commit.committer.date);
       const small_date = new Date(small.commit.committer.date);
       if (big_date > small_date) {
         return {
-          date : big_date,
-          sha : big.sha,
-          channel : "nixos-unstable",
+          date: big_date,
+          sha: big.sha,
+          channel: "nixos-unstable",
         };
       } else {
         return {
-          date : small_date,
-          sha : small.sha,
-          channel : "nixos-unstable-small",
+          date: small_date,
+          sha: small.sha,
+          channel: "nixos-unstable-small",
         };
       }
     });
   };
 
-  const url = get_revisions().then(get_newest).then(async (revision) => {
-    const next_sha = revision.sha;
-    const next_url =
-        `https://github.com/nixos/nixpkgs/archive/${next_sha}.tar.gz`;
-    const source_path = "./sources.nix";
+  function get_ocaml_commits(sha1, sha2, page = 1, prev_commits = []) {
```

Comment:
This function gets all the commits between the 2 shas and then filters the list so that we only keep commits starting with `ocaml`. It's a `function` because it needs to be recursive to get all commits since we can only get 100 at a time

Reply:
It'd be more interesting if we could get the result of `git ls-files` or something, rather than relying on commit messages. But this is a good first approximation.

Reply:
or even `git log --oneline SHA1...SHA2 -- pkgs/development/ocaml-modules`

Reply:
Yes, but then we need to clone the repo etc 😅 i don't think this information is easily available via the api